### ISSUE TITLE
Switch to more standardish/expected callback(error, data) pattern

### ISSUE THREAD

**Issue by jdub** [jdub/node-twitter#15]

**Comment by gasi** [comment]
+1

This issue and the mess the old `cookie` library creates in Express made me (temporarily) implement my own minimal Twitter client.

Have you considered taking advantage of some tools and libraries to make development easier and the code simpler?
Check out this implementation using [CoffeeScript](https://jashkenas.github.com/coffee-script) and [Streamline.js](https://github.com/Sage/streamlinejs):
https://github.com/gasi/node-twitter2/blob/develop/lib/twitter_.coffee

CoffeeScript gives us an abstraction for classes, string interpolation, etc. and Streamline really simplifies async code.

Let me know if you're interested in collaborating.

**Comment by jdub** [comment]
Sorry, but I'd like to stick to straight JS and callbacks for node-twitter. It'd be nice if you renamed your repo so no one is confused about the relationship between node-twitter and your own library. Thanks.

(The cookie issue is being resolved at the moment -- see the pull request.)

**Comment by gasi** [comment]
I respect your decision. Regarding the name, no worries, once your library doesn't rely on `cookie` and has standard callbacks, I'll probably kill mine. If I remember correctly, I read you mentioning to kill off the middleware & login part that relies on cookies. Is that true? I for one would welcome it very much.

Nice work on **node-twitter** :)

**Comment by jdub** [comment]
I'll move it to a different module (included with node-twitter or separate, haven't decided).

**Comment by gasi** [comment]
Cool, looking forward to check it out.

**Comment by ealdent** [comment]
+1 loving the library and this would be really nice to have

**Comment by jdub** [comment]
Check out the request branch for a first cut of this. Every post-0.1.18 release will use the standard node callback pattern. :-)

**Comment by aseemk** [comment]
Sweet! Looking forward to it. Thanks @jdub.

**Comment by jdub** [comment]
Just waiting on some fixes to request before merging the new request (and standard callback pattern) branch.

**Comment by gasi** [comment]
Here `json` is returned as first argument, even though it is the success case:
https://github.com/jdub/node-twitter/blob/5fa1af0d15f0d3499f7a4c3d6109f55e97a5b0d0/lib/twitter.js#L104

Is this an oversight or did you decide not to use the `callback(error, result)` pattern?

_P.S. Here I am, three years later, working for a different company and a different product and ran into the same issue. Made me chuckle seeing myself having commented on this back then_ :wink:

/cc @aseemk
